public/app/features/query/state/QueryRunner.ts

```diff
@@ -89,6 +89,11 @@ export class QueryRunner implements QueryRunnerSrv {
             return query;
           });
 
+          // Apply variables supported by the data source.
+          if (ds.interpolateVariablesInQueries) {
+            request.targets = ds.interpolateVariablesInQueries(request.targets, scopedVars ?? {});
+          }
```

Comment:
not sure we should add this, many data sources implement interpolateVariablesInQueries and they handle interpolation internally (usually after adding some custom scopedVars)

Reply:
I mean they handle interpolation in the query method, this would cause double interpolations

Reply:
Yeah, it looks like it is [always done by default](https://github.com/grafana/grafana/blob/f8faacd54a3ac1ac7ae523014c72f0830f562609/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L173) when executing `query()`. 

Hmm, our problem is that we would like to be able "share" the interpolated queries with the extensions, so they don't need to get hold of the datasource instance themselves to perform the interpolation.

Just as an idea - what if we would enrich the [`DataQueryReponse`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L439) on the frontend to contain the interpolated queries as well, and not just the return data? Would that be fundamentally wrong? What do you think @torkelo?

Reply:
> Hmm, our problem is that we would like to be able "share" the interpolated queries with the extensions, so they don't need to get hold of the datasource instance themselves to perform the interpolation.

How would the plugin use the query?

Reply:
> it's not always known which response comes from which query

Aaa, I see, I missed that, sorry. Hmm, I am not sure then, let me talk to @sd2k and @mckn today about how the query would actually be used by the extension.

Reply:
In the case of the ML plugin we use the query to prepopulate the 'Create forecast' form in the plugin - you can try it out by going to [this panel](https://play.grafana.org/d/000000029/prometheus-demo-dashboard?orgId=1&refresh=5m&viewPanel=2) and clicking 'Create Forecast' in the Extensions menu. Currently if the query contains template variables we end up with the non-interpolated query which we can't run.

Reply:
@torkelo thanks for great feedback! I will revert the interpolation part in the QueryRunner.

Reply:
> it's not always known which response comes from which query

I think Levi did a typo. So we don't need to interpolate the DataQueryResponse. It is the DataQuery we would like to interpolate.

Reply:
(I actually meant what if the `DataQueryResponse` that the plugin returns from the `query()` method would also include the interpolated queries, as in that case we would only need to run the interpolation once, inside of the `query()` method.)

Reply:
Ah, ok ok! My bad 🙈